src/main/java/org/gradlex/buildparameters/CodeGeneratingBuildParameter.java

```diff
@@ -26,12 +26,12 @@ interface CodeGeneratingBuildParameter {
 
     Identifier getId();
 
-    static CodeGeneratingBuildParameter from(BuildParameter<?> parameter) {
+    static CodeGeneratingBuildParameter from(BuildParameter<?> parameter, BuildParameterGroup containingGroup) {
         ParameterType type;
         if (parameter instanceof IntegerBuildParameter) {
             type = new ParameterType("int", "Integer", ".map(Integer::parseInt)", Function.identity());
         } else if (parameter instanceof BooleanBuildParameter) {
-            type = new ParameterType("boolean", "Boolean", ".map(Boolean::parseBoolean)", Function.identity());
+            type = new ParameterType("boolean", "Boolean", ".map(" + containingGroup.id.toSimpleTypeName() + "::parse" + parameter.id.toSimpleTypeName() + ")", Function.identity());
```

Comment:
At the moment the name of the parameter is generated into the method as part of the error message. If we have only one Util method, we would need to pass the name into that method as second parameter. Instead of 

`map(MyExtension::parseCi)` 

we would then generate something like

`map(p -> Util.parseBool(p, "ci"))` 

I kind of like that the static part is directly generated into the code where it is needed and we can then use the _slick_ `map(method::Reference)` syntax. And that we don't need to generate yet another class. Although these are probably not very strong arguments.

A better argument for the current solution is that it is all 'private'. So we can always change it later. If we add a `Util` class, which needs to be at least package visible, folks could theoretically write their own plugin code on top of the generated one and use the generated Util in there.

Reply:
Okay, I'm fine with the proposed solution 👍